components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -0,0 +1,1326 @@
+            coltend_cw_view,
+            // work arrays
+            raercol_cw_view, raercol_view, ekat::subview(nact, icol),
+            ekat::subview(mact, icol), ekat::subview(eddy_diff, icol),
+            ekat::subview(zn, icol), ekat::subview(csbot, icol),
+            ekat::subview(zs, icol), ekat::subview(overlapp, icol),
+            ekat::subview(overlapm, icol), ekat::subview(eddy_diff_kp, icol),
+            ekat::subview(eddy_diff_km, icol), ekat::subview(qncld, icol),
+            ekat::subview(srcn, icol), ekat::subview(source, icol),
+            ekat::subview(dz, icol), ekat::subview(csbot_cscen, icol),
+            ekat::subview(raertend, icol), ekat::subview(qqcwtend, icol));
+      });
+}
+
+// Update cloud borne aerosols
+void update_cloud_borne_aerosols(
+    haero::ThreadTeamPolicy team_policy,
+    const MAMAci::view_2d qqcw_fld_work[mam4::ndrop::ncnst_tot], const int nlev,
+    // output
+    mam_coupling::AerosolState &dry_aero) {
+  int ind_qqcw = 0;
+  for(int m = 0; m < mam_coupling::num_aero_modes(); ++m) {
+    Kokkos::deep_copy(dry_aero.cld_aero_nmr[m], qqcw_fld_work[ind_qqcw]);
+    ++ind_qqcw;
+    for(int a = 0; a < mam_coupling::num_aero_species(); ++a) {
+      if(dry_aero.cld_aero_mmr[m][a].data()) {
+        Kokkos::deep_copy(dry_aero.cld_aero_mmr[m][a], qqcw_fld_work[ind_qqcw]);
+        ++ind_qqcw;
+      }
+    }
+  }
+}
+
+// Update interstitial aerosols using tendencies - levels
+void update_interstitial_aerosols_levs(
+    const haero::ThreadTeam &team, const int nlev, const int icol,
+    const int s_idx, const Real dt,
+    const MAMAci::view_2d ptend_q[mam4::aero_model::pcnst],
+    // output
+    MAMAci::view_2d aero_mr) {
+  Kokkos::parallel_for(Kokkos::TeamThreadRange(team, nlev), [=](int kk) {
+    aero_mr(icol, kk) += ptend_q[s_idx](icol, kk) * dt;
+  });
+}
+
+// Update interstitial aerosols using tendencies- cols and levs
+void update_interstitial_aerosols(
+    haero::ThreadTeamPolicy team_policy,
+    const MAMAci::view_2d ptend_q[mam4::aero_model::pcnst], const int nlev,
+    const Real dt,
+    // output
+    mam_coupling::AerosolState &dry_aero) {
+  // starting index of ptend_q array (for MAM4, pcnst=40, ncnst_tot=25 )
+  int s_idx = mam4::aero_model::pcnst - mam4::ndrop::ncnst_tot;
+
+  // loop through all modes and species
+  for(int m = 0; m < mam_coupling::num_aero_modes(); ++m) {
+    for(int a = 0; a < mam4::num_species_mode(m); ++a) {
+      // mass mixing ratio of species "a" of mode "m"
+      auto aero_mmr = dry_aero.int_aero_mmr[m][a];
+
+      if(aero_mmr.data()) {
+        Kokkos::parallel_for(
+            team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+              const int icol = team.league_rank();
+              // update values for all levs at this column
+              update_interstitial_aerosols_levs(team, nlev, icol, s_idx, dt,
+                                                ptend_q,
```

Comment:
I am not 100% sure that it's ok to capture by value a static C array. I've had errors in the past when using them across a kernel boundary, so we should be extra safe and check with cuda compute-sanitizer as well as running on AMD gpus as well. Maybe I'm misremembering, or maybe it was a different use case. But just to be safe...

Reply:
The alternative to static arrays is to use a "View of Views", which is a bit annoying, but we already do that in Homme, for instance.

Reply:
I will fix it in a follow-up PR (see issue https://github.com/E3SM-Project/scream/issues/2850 )